### PR TITLE
Docker build setup

### PR DESCRIPTION
**Pull request by abrandemuehl**
Allows building for the host computer and the jetson

### CODE REVIEW

**Comment by abrandemuehl** [comment]
@shubhankar12 can you take a look at this. You should be able to use it even on your Mac if you install docker. Can you try it out?

Run `./build.sh host` from the root of your repository

File: tools/docker/jetson/get_debs.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    elif cache.has_key(package + ":all"):
+        pkg = cache[package]
+        # print(package + ": " + pkg.architecture())
+        final.append(pkg)
+    else:
+        # Ignore packages that aren't found. Usually this is because no versions
+        # of the package is published, meaning we can safely ignore it
+        print(package + " not found. Ignoring", file=sys.stderr)
+print('\n'.join([pkg.name for pkg in final]))
+
```
Comment (shubhankar12) [inline tools/docker/jetson/get_debs.py]: Do we need all these comments

File: tools/docker/jetson/get_debs.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+from __future__ import print_function
+
```
Comment (shubhankar12) [inline tools/docker/jetson/get_debs.py]: Can you add more documentation what this file is doing?

File: tools/docker/jetson/get_debs.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                elif cache.has_key(provider.name + ':all'):
+                    final.append(cache[provider.name + ':all'])
+                else:
+                    print(package + " not found for arm64 architecture. Ignoring", file=sys.stderr)
+        else:
+            arm_providers = [provider for provider in providers
+                             if provider.architecture() == 'arm64']
+            if len(arm_providers) == 1:
+                final.append(arm_providers[0])
+            elif len(arm_providers) > 1:
```
Comment (shubhankar12) [inline tools/docker/jetson/get_debs.py]: Is this the best way of doing it?
If it is the best way then Isn't the else if and else the same case?

File: tools/docker/jetson/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,61 @@
+FROM ubuntu:16.04
+
```
Comment (shubhankar12) [inline tools/docker/jetson/Dockerfile]: Add documentation of this file.

File: tools/docker/build.sh [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+set -e
+
```
Comment (shubhankar12) [inline tools/docker/build.sh]: Add more documentation about this file.
Differences in using host and jetson. 
Where to run it from.

File: tools/docker/jetson/get_debs.py (hunk 3 above)
Comment (abrandemuehl) [reply tools/docker/jetson/get_debs.py]: No, the else case is the length 0 case

File: README.md [hunk 6]
```diff
@@ -18,29 +18,9 @@ CI will check that the code builds and the tests pass. In addition, it will run
 ## Style
 We use clang-format to enforce style in the C++ files. You can format all of your code using `catkin_make --make-args clangformat`. It will automatically format all of your .cpp and .h files.
```
Comment (shubhankar12) [inline README.md]: You should mention what the ./build.sh host does. For the people using it in the future.
That it runs on mac as well with docker daemon running, will allow compiling robosub repository in docker thus allowing cross-compilation. Will not be able to execute the code. The main purpose is cross-compilation. In the future, you will be implementing syncing.
If you want you can mention the steps to run linter, ie. (pylint , clang-format).

**Comment by shubhankar12** [comment]
Tested that it runs on mac.

**Pull request merged**